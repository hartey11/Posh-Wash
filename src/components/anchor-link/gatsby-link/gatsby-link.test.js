import React from "react"
import { render, cleanup, fireEvent } from "@testing-library/react"
import GatsbyLink from "./gatsby-link.component"

afterEach(cleanup)

const clickEvent = jest.fn()
const title = "Some Title"
const href = "/this should encode correctly"

describe("GatsbyLink Component", () => {
  it("should render correctly", () => {
    const { asFragment } = render(
      <GatsbyLink isBrowser title={title} href={href}>
        Test
      </GatsbyLink>
    )
    expect(asFragment()).toMatchSnapshot()
  })

  it("should render with a clickEvent", () => {
    const { asFragment } = render(
      <GatsbyLink clickEvent={clickEvent} isBrowser title={title} href={href}>
        Test
      </GatsbyLink>
    )
    expect(asFragment()).toMatchSnapshot()
  })

  it("should trigger the click event correctly", () => {
    global.encodeURI = jest.fn()

    const { getByText } = render(
      <GatsbyLink clickEvent={clickEvent} isBrowser title={title} href={href}>
        Click Me
      </GatsbyLink>
    )

    fireEvent.click(getByText("Click Me"))

    expect(clickEvent).toHaveBeenCalled()
  })
})
