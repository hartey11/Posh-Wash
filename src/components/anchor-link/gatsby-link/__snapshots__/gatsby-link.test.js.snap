// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GatsbyLink Component should render correctly 1`] = `
<DocumentFragment>
  <a
    href="/this%20should%20encode%20correctly"
    title="Some Title"
  >
    Test
  </a>
</DocumentFragment>
`;

exports[`GatsbyLink Component should render with a clickEvent 1`] = `
<DocumentFragment>
  <a
    href="/this%20should%20encode%20correctly"
    title="Some Title"
  >
    Test
  </a>
</DocumentFragment>
`;
